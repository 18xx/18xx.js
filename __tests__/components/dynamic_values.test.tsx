import * as React from 'react';
import * as renderer from 'react-test-renderer';

import DynamicValues, {
  DynamicValueSet
} from '../../src/components/dynamic_values';

import Hexagon from '../../src/hexagon';

describe('DynamicValues', () => {
  const hexagon: Hexagon = new Hexagon('east-west');

  describe('#render', () => {
    const values: DynamicValueSet = { yellow: 20, green: 40 };

    describe('when fixedHeight is not set', () => {
      it('returns svg for the values', () => {
        const subject: React.ReactElement<DynamicValues> = (
          <DynamicValues hexagon={hexagon} values={values} />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when fixedHeight is set', () => {
      it('returns svg for the values', () => {
        const subject: React.ReactElement<DynamicValues> = (
          <DynamicValues fixedHeight={4} hexagon={hexagon} values={values} />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });
  });
});
