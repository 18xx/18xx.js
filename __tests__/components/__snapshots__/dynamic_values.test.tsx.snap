// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicValues #toString when fixedHeight is not set returns svg for the values 1`] = `"[object Object]"`;

exports[`DynamicValues #toString when fixedHeight is set returns svg for the values 1`] = `"[object Object]"`;
